src/vm/customattribute.cpp

```diff
@@ -166,8 +166,15 @@ CustomAttributeManagedValues Attribute::GetManagedCaValue(CaValue* pCaVal)
         BOOL bAllBlittableCa = arrayType != SERIALIZATION_TYPE_ENUM;
 
         if (length == (ULONG)-1)
+        {
+            GCPROTECT_BEGIN(gc)
+            {   
+                if (arrayType == SERIALIZATION_TYPE_ENUM)
```

Comment:
>Sorry, I'm a bit lost here... there is a first  return gc; 

Ah yes, that "return gc" will screw things up. Yes, we should delete that and use and turn the "if" into an "if {...} else {...}". Good catch.